app/controllers/subject_rules_controller.rb

```diff
@@ -0,0 +1,72 @@
+class SubjectRulesController < ApplicationController
+  def index
+    authorize workflow
+
+    rules = SubjectRule.where(workflow_id: params[:workflow_id])
+    respond_to do |format|
+      format.html { redirect_to workflow }
+      format.json { render json: rules }
+    end
+  end
+
+  def show
+    authorize workflow
+    @rule = workflow.subject_rules.find(params[:id]) or not_found
```

Comment:
Did you TDD the addition of `or not_found`? If I remove it, I get no test failures, and I'm fairly sure that it's unreachable code since Rails would raise `RecordNotFound`. This in turn is handled by the ApplicationController, which is why you are seeing a 404. Since `or` tends to bite, and it doesn't have any effect here, perhaps these could just be removed everywhere? (I won't mark all of them)

Reply:
yeah, let's just get rid of them in that case--i thought they were the ones producing the 404 i wanted, but i didn't write the failing test first